### ISSUE TITLE
ToListAsync() is very slow

### ISSUE THREAD

**Issue by lukwag** [dotnet/efcore#18571]
### Problem
If entities that contains large strings are loaded with ToList() every thing will work fine. But if the same entities are loaded with ToListAsync() the CPU utilisation and the statement execution will become a nightmare.

I have already found some information about this problem on stack overflow ([Entity Framework async operation takes ten times as long to complete](https://stackoverflow.com/questions/28543293/entity-framework-async-operation-takes-ten-times-as-long-to-complete)) but I'm still looking for a solution to solve this problem.

### Steps to reproduce
I have added a sample project to reproduce this behavouir.
The sample project contains a console application that just creates 5 records that contains a field with a large string (1 MB) and selects this records 10 times with ToList() and 10 times with ToListAsync(). The difference in execution times is huge.

If you change the size of the test string from 1 MB to e.g. 3 or 4 MB the select query will become a never ending story with the ToListAsync() function.

### Sample project
[EFCoreAsyncPerformanceTest.zip](https://github.com/aspnet/EntityFrameworkCore/files/3767248/EFCoreAsyncPerformanceTest.zip)

### Technical details
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 3.0
IDE: Visual Studio 2019

**Comment by AndriySvyryd** [comment]
This is a duplicate of https://github.com/aspnet/EntityFrameworkCore/issues/885, blocked by [this issue in SqlClient](https://github.com/dotnet/SqlClient/issues/245)

Workarounds: Don't use `VARCHAR(MAX)` or don't use async queries.
